en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_ReactJS_Redux_and_GraphQL.md

```diff
@@ -809,68 +810,162 @@ You could manipulate the action called by the originalReducer, there isn't an ex
 ```
-Lastly, we'll expose the model to GraphQL:
+
+Next we'll expose the model to GraphQL:
+
+#### Graphql v4 {#build-extensible-gql-app-v4}
+
+**my-module/_config/graphql.yml**
 
 ```yml
-SilverStripe\GraphQL\Controller:
-  schema:
-    scaffolding:
-      types:
-        MyApp\AwesomeNotes\Note:
-          fields: [ID, Content]
-          operations:
-            read:
-              paginate: false
-            create: true
-            
+# Tell graphql that we're adding to the admin graphql schema
+SilverStripe\GraphQL\Schema\Schema:
+  schemas:
+    admin:
+      src:
+        - my-module/_graphql
+```
+
+**my-module/_graphql/models.yml**
+
+```yml
+# Tell graphql how to scaffold the schema for our model
+App\Model\Note:
+  fields:
+    id: true
+    content: true
+  operations:
+    read:
+      plugins:
+        paginateList: false
+```
+
+#### Graphql v3 {#build-extensible-gql-app-v3}
+
+**my-module/_config/graphql.yml**
+
+```yml
+# Tell graphql how to scaffold the schema for our model inside the admin schema
+SilverStripe\GraphQL\Manager:
+  schemas:
+    admin:
+      scaffolding:
+        types:
+          App\Model\Note:
+            fields: [ id, content ]
+            operations:
+              read:
+                paginate: false
+                name: readNotes
+              create: true
+```
+
+[hint]
+Graphql v3 uses the first part of the model class's namespace in the default query/mutation names - so with a class `App\Model\Note` the default read operation would be `readAppNotes`. The example above has overridden the default name to keep it consistent with Graphql v4 behaviour.
+[/hint]
+
+#### Define the app
+
+Finally, let's make a really simple container app which holds a header and our notes component, and inject it into the DOM using entwine.
+
+**my-module/client/src/App.js**
+
+```js
+import React from 'react';
+import { inject } from 'lib/Injector';
+import Notes from './components/Notes';
+
+const App = ({ ListComponent }) => (
+  <div>
+    <h3>Notes</h3>
+    <Notes />
+  </div>
+);
+```
+
+**my-module/client/src/index.js**
+```js
+import ReactDOM from 'react-dom';
+import React from 'react';
+import { ApolloProvider } from 'react-apollo';
+import Injector from 'lib/Injector';
+import App from './App';
+
+Injector.ready(() => {
+  const { apolloClient, store } = window.ss;
+
+  // Assuming you've got some element in the DOM with the id "notes-app"
+  $('#notes-app').entwine({
+    onmatch() {
+      ReactDOM.render(
+        <ApolloProvider client={apolloClient} store={store}>
+          <App />
+        </ApolloProvider>,
+        this[0]
+      )
+    },
+
+    onunmatch: function() {
+      ReactDOM.unmountComponentAtNode(this[0]);
+    }
+  })
+});
 ```
 
+The `silverstripe/admin` module was generous enough to hang its `apolloClient` and `store` objects in the global namespace to be shared by other modules. We'll make use of those, and create our own app wrapped in `<ApolloProvider />`.
```

Comment:
```suggestion
The `silverstripe/admin` module provides `apolloClient` and `store` objects in the global namespace to be shared by other modules. We'll make use of those, and create our own app wrapped in `<ApolloProvider />`.
```

Reply:
Oops I thought I'd made this change but looks like I forgot. Done now.